### DISCUSSION TITLE
Android 6 will be retired soon

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#3491] [Show and tell]
I am sorry that we have no choice but to retire Android 6.  
We need to do that so that xDrip can be used on Android 15.

This will happen in an upcoming Nightly release.

We have tried to delay this as much as possible so that no one is forced to have to spend money to run xDrip.
But, unfortunately, because of the changes made by Google, we have no choice.

**Comment by PhilTU** [comment]
Is this because there is a limit on how many versions can be suipported at one time ?

**Reply by Navid200** [reply]
I'm not sure I understand what you are thinking.
I am running Android 15 beta on my phone and cannot install xDrip as is.
This PR will change the target Android and I am using that now: https://github.com/NightscoutFoundation/xDrip/pull/3483
But, that PR can not be used on Android 6.

I don't want to speculate.  Please tell me exactly what you are asking.

**Reply by PhilTU** [reply]
I am asking out of curiosity if dropping 6 is necessary because there are issues with 6 or because in order to accommodate 15 you need to drop 6 because you can only support 7-15 ie 8 versions at once (seems possible, as a power of 2)

**Reply by Navid200** [reply]
No, there is no set rule in that sense as fart as I know.

Google decides what is in a new OS.  What apps can or cannot be installed in Android 15 is enforced by Android 15 (Google).
If an app targets Android 6 or older, Android 15 will not allow it to be installed.

We retired Android 4 in October of 2022.  Android 13 had already been released then.  This means that Android 13 allowed apps targeting Android 4 to be installed.  
Google is gradually becoming more and more restrictive in that sense unfortunately.

Android 14 complained about installing xDrip.  It still does.  You have to tap on install anyway to get it installed.
Even after we retired Android 6, you will still need to tap on install anyway to install xDrip on Android 15.
So, it's Google that is forcing us to retire older releases sooner and sooner.

**Reply by PhilTU** [reply]
Ok. I thought the "install anyway" thing arose just because it's coming from an apk - I get that on Android 9.

**Reply by Navid200** [reply]
You are right.  The install anyway requirement I believe comes from Google Play Protect.  

But, even if I disable Google Play Protect, I cannot install the current Nightly on Android 15 beta 2.

**Reply by parapenT1sta** [reply]
> You are right. The install anyway requirement I believe comes from Google Play Protect.
> 
> But, even if I disable Google Play Protect, I cannot install the current Nightly on Android 15 beta 2.

Just curiosity, did you test running the adb command: _adb install --bypass-low-target-sdk-block xdrip.apk_ ? I would like to know if it still works on Android 15.

**Reply by Navid200** [reply]
No. I didn't.
Please test it and tell us how it goes.

**Reply by parapenT1sta** [reply]
> No. I didn't. Please test it and tell us how it goes.

I can't test because I only have Android 14.
